nameko/messaging.py

```diff
@@ -331,7 +328,10 @@ def connection(self):
             return   # don't cache a connection during introspection
 
         if self._connection is None:
-            self._connection = Connection(self.amqp_uri)
+            heartbeat = self.container.config.get(
+                HEARTBEAT_CONFIG_KEY, DEFAULT_HEARTBEAT
+            )
+            self._connection = Connection(self.amqp_uri, heartbeat=heartbeat)
```

Comment:
There is some extra network traffic but no performance implication as far as I'm aware. Services will probably get faster because broken connections are discovered earlier.